### PR TITLE
Add `find_note_def` to find the definition(s) of a Note

### PR DESCRIPTION
**Pull request by sgraf812**
Examples:
```bash
$ cd /path/to/ghc/checkout
$ find_note_def "Precise exceptions and Strictness analysis"
compiler/GHC/Types/Demand.hs:1548
$ echo $?
0
$ find_note_def Capital
$ echo $?
42
$ find_note_def Shadowing
compiler/GHC/Core/Opt/CSE.hs:56
compiler/GHC/Core/Opt/Simplify/Iteration.hs:123
compiler/GHC/Core/Opt/Simplify/Iteration.hs:2410
compiler/GHC/Core/Opt/SpecConstr.hs:267
compiler/GHC/Core.hs:316
$ echo $?
42
```

It uses ripgrep, so it's blazing fast. Using this works great for tooling.

### CODE REVIEW

File: ghc.nix [hunk 1]
```diff
@@ -165,6 +175,26 @@ let
         ];
         librarySystemDepends = depsSystem;
       });
+
+  findNoteDef = writeShellScriptBin "find_note_def" ''
+    ret=$(rg  --no-messages --vimgrep -i --engine pcre2 "^ ?[{\\-#*]* *\QNote [$1]\E\s*$")
```
Comment (MangoIV) [inline ghc.nix:180]: I think it would be preferable to refer directly to the nixpkgs version here eg `${pkgs.lib.getExe pkgs.ripgrep}`

Comment (sgraf812) [reply ghc.nix:180]: Oh, good catch. This is an oversight from my local experiments... What's the difference between `${pkgs.ripgrep}/bin/rg` and `${pkgs.lib.getExe pkgs.ripgrep}`?

File: README.md [hunk 2]
```diff
@@ -194,6 +194,7 @@ be careful to specify the path to the `shell.nix`, not to the `default.nix`.
 | `withGrind` | whether to include `valgrind` | `true` | ❌ |
 | `withEMSDK` | whether to include `emscripten` for the js-backend, will create an `.emscripten_cache` folder in your working directory of the shell for writing. `EM_CACHE` is set to that path, prevents [sub word sized atomic](https://gitlab.haskell.org/ghc/ghc/-/wikis/javascript-backend/building#configure-fails-with-sub-word-sized-atomic-operations-not-available) kinds of issues | `false` | ❌ |
 | `withWasiSDK` | whether to include `wasi-sdk` & `wasmtime` for the ghc wasm backend | `false` | ❌ |
+| `withFindNoteDef` | install a shell script `find_note_def`; `find_note_def "Adding a language extension"` will point to the definition of the Note "Adding a language extension" | `true` | ✅ |
```
Comment (MangoIV) [inline README.md:197]: Last minor nit:
It's not a flake input so it should actually be an here ❌

**Review by MangoIV — APPROVED** [review]

**Review by supersven — APPROVED** [review]
LGTM :+1:

**Pull request merged**